src/pwm/pwm.c

```diff
@@ -61,7 +61,10 @@ mraa_pwm_write_period(mraa_pwm_context dev, int period)
         mraa_result_t result = dev->advance_func->pwm_period_replace(dev, period);
         if (result == MRAA_SUCCESS) {
             dev->period = period;
+        } else if (result == MRAA_ERROR_FEATURE_NOT_IMPLEMENTED) {
+            return MRAA_SUCCESS;
```

Comment:
The return here should be MRAA_ERROR_FEATURE_NOT_IMPLEMENTED passed by pwm.c, I don't see how anything else is appropriate. UPM drivers trying to use this with firmata will have to check/handle it.